Include ansible playbook for test-node-eviction-out-of-disk.

dargasudarshan opened the pull request:
Signed-off-by: Sudarshan Darga <sudarshan.darga@cloudbyte.com>

<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:
- Includes an ansible playbook test suite to induce taint with Eviction Condition and verify the behavior.
- Configure node affinity to stick pod to a specified node.
- Deploy application with replica's scheduled as per node affinity policy.
- Simulate node evictions by tainting a node with condition "out-of-disk".
> kubectl taint nodes kubeminion-01 node.kubernetes.io/memory-pressure=:NoExecute
- Verify that pod scheduled on tainted node is evicted as per taint condition.
- Also verify that pod is not scheduled on any other nodes as per node affinity policy set.
- Remove the simulated eviction taint condition.
> kubectl taint nodes kubeminion-01 node.kubernetes.io/memory-pressure:NoExecute-
- Verify that pods are scheduled and running successfully again.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
@ksatchit Can you please review this PR?

dargasudarshan commented:
Lint Output:

dargasudarshan commented:
devops@devops-virtual-machine:~/PR6/openebs$ ansible-lint e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml

ksatchit commented on e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk-vars.yml:
```diff
@@ -0,0 +1,26 @@
+
+percona_link: https://raw.githubusercontent.com/openebs/openebs/master/k8s/demo/percona/demo-percona-mysql-pvc.yaml
+
+replica_count: 2 
```
Can we arrive at this dynamically ? i.e., get the node count and decrease by 1 to obtain the replica count.

dargasudarshan replied:
Yeah, I will do that.

dargasudarshan replied:
Done @ksatchit Can you please have a look?

dargasudarshan commented:
@yudaykiran Playbook o/p with latest changes:

```devops@devops-virtual-machine:~/PR6/openebs/e2e/ansible$ ansible-playbook playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml -vv
ansible-playbook 2.4.3.0
  config file = /home/devops/PR6/openebs/e2e/ansible/ansible.cfg
  configured module search path = [u'/home/devops/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
Using /home/devops/PR6/openebs/e2e/ansible/ansible.cfg as config file
[DEPRECATION WARNING]: The use of 'include' for tasks has been deprecated. Use 'import_tasks' for static inclusions or
'include_tasks' for dynamic inclusions. This feature will be removed in a future release. Deprecation warnings can be disabled by
setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: include is kept for backwards compatibility but usage is discouraged. The module documentation details page
may explain more about this rationale.. This feature will be removed in a future release. Deprecation warnings can be disabled by
setting deprecation_warnings=False in ansible.cfg.
statically imported: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk-prereq.yml
statically imported: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk-cleanup.yml

PLAYBOOK: test-node-eviction-out-of-disk.yml *****************************************************************************************
1 plays in playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml

PLAY [localhost] *********************************************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************
ok: [localhost]
META: ran handlers

TASK [Get $HOME of K8s master for kubernetes user] ***********************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk-prereq.yml:2
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; echo $HOME", "delta": "0:00:00.003672", "end": "2018-05-11 11:42:18.106982", "rc": 0, "start": "2018-05-11 11:42:18.103310", "stderr": "", "stderr_lines": [], "stdout": "/home/ubuntu", "stdout_lines": ["/home/ubuntu"]}

TASK [Start the log aggregator to capture test pod logs] *****************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk-prereq.yml:9
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; nohup stern \"maya*|openebs*|pvc*|percona*\" --since 1m > \"/home/ubuntu/setup/logs/test-node-eviction-out-of-disk.log\" &", "delta": "0:00:00.005898", "end": "2018-05-11 11:42:18.448794", "rc": 0, "start": "2018-05-11 11:42:18.442896", "stderr": "mesg: ttyname failed: Inappropriate ioctl for device\n/bin/bash: /home/ubuntu/setup/logs/test-node-eviction-out-of-disk.log: No such file or directory", "stderr_lines": ["mesg: ttyname failed: Inappropriate ioctl for device", "/bin/bash: /home/ubuntu/setup/logs/test-node-eviction-out-of-disk.log: No such file or directory"], "stdout": "", "stdout_lines": []}

TASK [include_tasks] *****************************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:33
included: /home/devops/openebs/e2e/ansible/playbooks/utils/deploy_check.yml for localhost

TASK [Check maya-api pod status] *****************************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/utils/deploy_check.yml:2
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n openebs | grep maya-api", "delta": "0:00:00.298952", "end": "2018-05-11 11:42:19.104496", "rc": 0, "start": "2018-05-11 11:42:18.805544", "stderr": "", "stderr_lines": [], "stdout": "maya-apiserver-6d945596b-lsdgm         1/1       Running   0          1h", "stdout_lines": ["maya-apiserver-6d945596b-lsdgm         1/1       Running   0          1h"]}

TASK [1b) Obtaining storage classes yaml] ********************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:38
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get sc openebs-percona -o yaml > \"storage-class.yaml\"", "delta": "0:00:00.234225", "end": "2018-05-11 11:42:19.656481", "rc": 0, "start": "2018-05-11 11:42:19.422256", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [1c) Get the Number of nodes count] *********************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:44
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get nodes | grep 'Ready' | grep 'none' | wc -l", "delta": "0:00:00.214812", "end": "2018-05-11 11:42:20.182488", "rc": 0, "start": "2018-05-11 11:42:19.967676", "stderr": "", "stderr_lines": [], "stdout": "3", "stdout_lines": ["3"]}

TASK [Fetch the node count from stdout] **********************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:51
ok: [localhost] => {"ansible_facts": {"node_count": " 3"}, "changed": false}

TASK [1d) Replace the replica count in storage classes yaml] *************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:55
ok: [localhost -> None] => {"changed": false, "msg": ""}

TASK [2a) Delete the existing storage class and create new one] **********************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:68
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "kubectl delete sc openebs-percona; kubectl apply -f \"storage-class.yaml\"", "delta": "0:00:00.617086", "end": "2018-05-11 11:42:21.618009", "rc": 0, "start": "2018-05-11 11:42:21.000923", "stderr": "", "stderr_lines": [], "stdout": "storageclass \"openebs-percona\" deleted\nstorageclass \"openebs-percona\" created", "stdout_lines": ["storageclass \"openebs-percona\" deleted", "storageclass \"openebs-percona\" created"]}

TASK [Create nodeeviction namespace] *************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:78
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "kubectl create ns nodeeviction", "delta": "0:00:00.195124", "end": "2018-05-11 11:42:22.113175", "rc": 0, "start": "2018-05-11 11:42:21.918051", "stderr": "", "stderr_lines": [], "stdout": "namespace \"nodeeviction\" created", "stdout_lines": ["namespace \"nodeeviction\" created"]}

TASK [include_tasks] *****************************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:88
included: /home/devops/openebs/e2e/ansible/playbooks/utils/deploy_task.yml for localhost

TASK [Deploy application files] ******************************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/utils/deploy_task.yml:2
changed: [localhost -> None] => (item=https://raw.githubusercontent.com/openebs/openebs/master/k8s/demo/percona/demo-percona-mysql-pvc.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f https://raw.githubusercontent.com/openebs/openebs/master/k8s/demo/percona/demo-percona-mysql-pvc.yaml -n nodeeviction", "delta": "0:00:01.489793", "end": "2018-05-11 11:42:23.976249", "item": "https://raw.githubusercontent.com/openebs/openebs/master/k8s/demo/percona/demo-percona-mysql-pvc.yaml", "rc": 0, "start": "2018-05-11 11:42:22.486456", "stderr": "", "stderr_lines": [], "stdout": "pod \"percona\" created\npersistentvolumeclaim \"demo-vol1-claim\" created", "stdout_lines": ["pod \"percona\" created", "persistentvolumeclaim \"demo-vol1-claim\" created"]}

TASK [include_tasks] *****************************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:93
included: /home/devops/openebs/e2e/ansible/playbooks/utils/deploy_check.yml for localhost

TASK [Check percona pod status] ******************************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/utils/deploy_check.yml:2
FAILED - RETRYING: Check percona pod status (15 retries left).
FAILED - RETRYING: Check percona pod status (14 retries left).
FAILED - RETRYING: Check percona pod status (13 retries left).
FAILED - RETRYING: Check percona pod status (12 retries left).
FAILED - RETRYING: Check percona pod status (11 retries left).
changed: [localhost -> None] => {"attempts": 6, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n nodeeviction | grep percona", "delta": "0:00:00.274023", "end": "2018-05-11 11:44:58.260666", "rc": 0, "start": "2018-05-11 11:44:57.986643", "stderr": "", "stderr_lines": [], "stdout": "percona                                                          1/1       Running   0          2m", "stdout_lines": ["percona                                                          1/1       Running   0          2m"]}

TASK [2c) Check specified number of replica pods are created] ************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:98
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: '{{ (node_count) |int-1
}}' in rep_count.stdout

changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "kubectl get pods --all-namespaces | grep rep | grep -i running |wc -l", "delta": "0:00:00.288278", "end": "2018-05-11 11:44:58.824305", "rc": 0, "start": "2018-05-11 11:44:58.536027", "stderr": "", "stderr_lines": [], "stdout": "2", "stdout_lines": ["2"]}

TASK [Get pv name associated with percona application] *******************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:109
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get pv | grep openebs-percona | awk '{print $1}'", "delta": "0:00:00.210956", "end": "2018-05-11 11:44:59.322304", "rc": 0, "start": "2018-05-11 11:44:59.111348", "stderr": "", "stderr_lines": [], "stdout": "pvc-5f300bd9-5510-11e8-9653-02b983f0a4db", "stdout_lines": ["pvc-5f300bd9-5510-11e8-9653-02b983f0a4db"]}

TASK [include_tasks] *****************************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:116
included: /home/devops/openebs/e2e/ansible/playbooks/utils/copy_task.yml for localhost

TASK [Copy the specs/files to kubemaster] ********************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/utils/copy_task.yml:2
changed: [localhost -> None] => (item=replica_patch.yaml) => {"changed": true, "checksum": "6c1b0a07601d8a40ec5b7d8b66fabd310d1d47c9", "dest": "/home/ubuntu/replica_patch.yaml", "gid": 1000, "group": "ubuntu", "item": "replica_patch.yaml", "md5sum": "00730266803f0063024795e2cb628c1b", "mode": "0664", "owner": "ubuntu", "size": 406, "src": "/home/ubuntu/.ansible/tmp/ansible-tmp-1526039099.56-252927888131264/source", "state": "file", "uid": 1000}

TASK [2e) Gather node details for scheduled replicas] ********************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:120
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get pod -n nodeeviction -o wide | grep \"pvc-5f300bd9-5510-11e8-9653-02b983f0a4db\" | grep rep | awk '{print $7}'", "delta": "0:00:00.264021", "end": "2018-05-11 11:45:00.936735", "rc": 0, "start": "2018-05-11 11:45:00.672714", "stderr": "", "stderr_lines": [], "stdout": "kubeminion-02\nkubeminion-03", "stdout_lines": ["kubeminion-02", "kubeminion-03"]}

TASK [include_tasks] *****************************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:127
included: /home/devops/openebs/e2e/ansible/playbooks/utils/regex_task.yml for localhost

TASK [Replace volume-claim name with test parameters] ********************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/utils/regex_task.yml:2
changed: [localhost -> None] => (item=[u'nodename_where_replica_pod_1_got_scheduled', u'kubeminion-02']) => {"changed": true, "item": ["nodename_where_replica_pod_1_got_scheduled", "kubeminion-02"], "msg": "1 replacements made"}
changed: [localhost -> None] => (item=[u'nodename_where_replica_pod_2_got_scheduled', u'kubeminion-03']) => {"changed": true, "item": ["nodename_where_replica_pod_2_got_scheduled", "kubeminion-03"], "msg": "1 replacements made"}

TASK [3a) Get deployment details to set node affinity] *******************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:133
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get deploy -n nodeeviction |grep \"pvc-5f300bd9-5510-11e8-9653-02b983f0a4db\" | grep rep |awk '{print $1}'", "delta": "0:00:00.281785", "end": "2018-05-11 11:45:02.184373", "rc": 0, "start": "2018-05-11 11:45:01.902588", "stderr": "", "stderr_lines": [], "stdout": "pvc-5f300bd9-5510-11e8-9653-02b983f0a4db-rep", "stdout_lines": ["pvc-5f300bd9-5510-11e8-9653-02b983f0a4db-rep"]}

TASK [Wait for 120s before applying patch] *******************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:140
ok: [localhost] => {"changed": false, "elapsed": 120, "path": null, "port": null, "search_regex": null, "state": "started"}

TASK [3b) Set replica deployment with node affinity policy] **************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:144
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "kubectl patch deployment \"pvc-5f300bd9-5510-11e8-9653-02b983f0a4db-rep\" -n nodeeviction -p \"$(cat replica_patch.yaml)\"", "delta": "0:00:00.253285", "end": "2018-05-11 11:47:04.304933", "rc": 0, "start": "2018-05-11 11:47:04.051648", "stderr": "", "stderr_lines": [], "stdout": "deployment \"pvc-5f300bd9-5510-11e8-9653-02b983f0a4db-rep\" patched", "stdout_lines": ["deployment \"pvc-5f300bd9-5510-11e8-9653-02b983f0a4db-rep\" patched"]}

TASK [3c) Check whether replica pods are re-scheduled and running] *******************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:154
FAILED - RETRYING: 3c) Check whether replica pods are re-scheduled and running (15 retries left).
FAILED - RETRYING: 3c) Check whether replica pods are re-scheduled and running (14 retries left).
changed: [localhost -> None] => {"attempts": 3, "changed": true, "cmd": "kubectl get pods -n nodeeviction | grep rep | grep -i running |wc -l", "delta": "0:00:00.322586", "end": "2018-05-11 11:48:06.382590", "rc": 0, "start": "2018-05-11 11:48:06.060004", "stderr": "", "stderr_lines": [], "stdout": "2", "stdout_lines": ["2"]}

TASK [4a) Get node on which replica is scheduled] ************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:171
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get pods -n nodeeviction -o wide | grep rep | awk 'FNR == 1 {print}' | awk {'print $7'}", "delta": "0:00:00.330387", "end": "2018-05-11 11:48:07.064266", "rc": 0, "start": "2018-05-11 11:48:06.733879", "stderr": "", "stderr_lines": [], "stdout": "kubeminion-02", "stdout_lines": ["kubeminion-02"]}

TASK [4b) Taint node with Eviction Condition node.kubernetes.io/out-of-disk] *********************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:178
 [WARNING]: As of Ansible 2.4, the parameter 'executable' is no longer supported with the 'command' module. Not using '/bin/bash'.

changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": ["kubectl", "taint", "nodes", "kubeminion-02", "node.kubernetes.io/out-of-disk=:NoExecute"], "delta": "0:00:00.263818", "end": "2018-05-11 11:48:07.659479", "rc": 0, "start": "2018-05-11 11:48:07.395661", "stderr": "", "stderr_lines": [], "stdout": "node \"kubeminion-02\" tainted", "stdout_lines": ["node \"kubeminion-02\" tainted"]}

TASK [4c) Verify replica pod remains in Pending state] *******************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:188
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n nodeeviction | grep pvc | grep -i Pending |wc -l", "delta": "0:00:00.441727", "end": "2018-05-11 11:48:08.612704", "rc": 0, "start": "2018-05-11 11:48:08.170977", "stderr": "", "stderr_lines": [], "stdout": "1", "stdout_lines": ["1"]}

TASK [5a) Get pod_name to Verify replica pod is scheduled on same node] **************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:205
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get pods -n nodeeviction | grep pvc | grep -i Pending | awk {'print $1'}", "delta": "0:00:00.296160", "end": "2018-05-11 11:48:09.243560", "rc": 0, "start": "2018-05-11 11:48:08.947400", "stderr": "", "stderr_lines": [], "stdout": "pvc-5f300bd9-5510-11e8-9653-02b983f0a4db-rep-f947f5f57-2kj5h", "stdout_lines": ["pvc-5f300bd9-5510-11e8-9653-02b983f0a4db-rep-f947f5f57-2kj5h"]}

TASK [5b) Untaint node with Eviction Condition node.kubernetes.io/out-of-disk] *******************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:212
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": ["kubectl", "taint", "nodes", "kubeminion-02", "node.kubernetes.io/out-of-disk:NoExecute-"], "delta": "0:00:00.278621", "end": "2018-05-11 11:48:09.871243", "rc": 0, "start": "2018-05-11 11:48:09.592622", "stderr": "", "stderr_lines": [], "stdout": "node \"kubeminion-02\" untainted", "stdout_lines": ["node \"kubeminion-02\" untainted"]}

TASK [5c) Verify replica pod is scheduled on same node adhering NodeAffinity Policy] *************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:222
FAILED - RETRYING: 5c) Verify replica pod is scheduled on same node adhering NodeAffinity Policy (5 retries left).
FAILED - RETRYING: 5c) Verify replica pod is scheduled on same node adhering NodeAffinity Policy (4 retries left).
changed: [localhost -> None] => {"attempts": 3, "changed": true, "cmd": "kubectl get pods -n nodeeviction -o wide | grep kubeminion-02 | grep pvc-5f300bd9-5510-11e8-9653-02b983f0a4db-rep-f947f5f57-2kj5h", "delta": "0:00:00.260852", "end": "2018-05-11 11:49:11.884191", "rc": 0, "start": "2018-05-11 11:49:11.623339", "stderr": "", "stderr_lines": [], "stdout": "pvc-5f300bd9-5510-11e8-9653-02b983f0a4db-rep-f947f5f57-2kj5h     1/1       Running       0          1m        10.36.0.3   kubeminion-02", "stdout_lines": ["pvc-5f300bd9-5510-11e8-9653-02b983f0a4db-rep-f947f5f57-2kj5h     1/1       Running       0          1m        10.36.0.3   kubeminion-02"]}

TASK [Test Passed] *******************************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:232
ok: [localhost] => {"ansible_facts": {"flag": "Pass"}, "changed": false}

TASK [include_tasks] *****************************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk-cleanup.yml:2
included: /home/devops/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml for localhost

TASK [Deploy application files] ******************************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml:2
changed: [localhost -> None] => (item=https://raw.githubusercontent.com/openebs/openebs/master/k8s/demo/percona/demo-percona-mysql-pvc.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl delete -f https://raw.githubusercontent.com/openebs/openebs/master/k8s/demo/percona/demo-percona-mysql-pvc.yaml -n nodeeviction", "delta": "0:00:01.514020", "end": "2018-05-11 11:49:13.799339", "item": "https://raw.githubusercontent.com/openebs/openebs/master/k8s/demo/percona/demo-percona-mysql-pvc.yaml", "rc": 0, "start": "2018-05-11 11:49:12.285319", "stderr": "", "stderr_lines": [], "stdout": "pod \"percona\" deleted\npersistentvolumeclaim \"demo-vol1-claim\" deleted", "stdout_lines": ["pod \"percona\" deleted", "persistentvolumeclaim \"demo-vol1-claim\" deleted"]}

TASK [include_tasks] *****************************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk-cleanup.yml:7
 [WARNING]: Ignoring invalid attribute: ignore_erros

included: /home/devops/openebs/e2e/ansible/playbooks/utils/delete_deploy_check.yml for localhost

TASK [Check pod status] **************************************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/utils/delete_deploy_check.yml:2
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n nodeeviction | grep percona", "delta": "0:00:00.317761", "end": "2018-05-11 11:49:14.546945", "rc": 0, "start": "2018-05-11 11:49:14.229184", "stderr": "", "stderr_lines": [], "stdout": "percona                                                          1/1       Terminating   0          6m", "stdout_lines": ["percona                                                          1/1       Terminating   0          6m"]}

TASK [Delete nodeeviction ns] ********************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk-cleanup.yml:12
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "kubectl delete ns nodeeviction", "delta": "0:00:00.219060", "end": "2018-05-11 11:49:15.110743", "rc": 0, "start": "2018-05-11 11:49:14.891683", "stderr": "", "stderr_lines": [], "stdout": "namespace \"nodeeviction\" deleted", "stdout_lines": ["namespace \"nodeeviction\" deleted"]}

TASK [Confirm pvc pod has been deleted] **********************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk-cleanup.yml:22
FAILED - RETRYING: Confirm pvc pod has been deleted (10 retries left).
fatal: [localhost -> None]: FAILED! => {"attempts": 2, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n nodeeviction | grep pvc", "delta": "0:00:00.314293", "end": "2018-05-11 11:49:46.452261", "msg": "non-zero return code", "rc": 1, "start": "2018-05-11 11:49:46.137968", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
...ignoring

TASK [Remove copied files] ***********************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk-cleanup.yml:33
 [WARNING]: Consider using file module with state=absent rather than running rm

changed: [localhost] => {"changed": true, "cmd": "rm -rf \"replica_patch.yaml\" \"storage-class.yaml\"", "delta": "0:00:00.008170", "end": "2018-05-11 17:19:46.690220", "rc": 0, "start": "2018-05-11 17:19:46.682050", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [Test Cleanup Passed] ***********************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:246
ok: [localhost] => {"ansible_facts": {"cflag": "cleanup_pass"}, "changed": false}

TASK [6) Send slack notification] ****************************************************************************************************
task path: /home/devops/PR6/openebs/e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:264
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional result was False"}
META: ran handlers
META: ran handlers

PLAY RECAP ***************************************************************************************************************************
localhost                  : ok=41   changed=28   unreachable=0    failed=0
```

yudaykiran commented on e2e/ansible/playbooks/resiliency/test-node-eviction-out-of-disk/test-node-eviction-out-of-disk.yml:
```diff
@@ -260,11 +254,11 @@
-             args:
-               executable: /bin/bash
-             delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+           #- name: 5) Terminate the log aggregator
```
Remove the code if it is not being used. Comments only for help.
